### ISSUE TITLE
class instantiation failing

### ISSUE THREAD

**Issue by harshal-shah** [pcfens/puppet-filebeat#4]
Hi, 
I downloaded the module for filebeat and tried the default class declaration as below: 

node 'fbes1.hs.com'{
	include filebeat
	class { 'filebeat':
  outputs => {
    'logstash' => {
     'hosts' => ['lsredis.hs.com:5044',],
     'loadbalance' => true,
    },
  },
}
}

but the class declaration fails with below error: 

[root@fbes1 ~]# puppet agent -t --noop
Info: Retrieving pluginfacts
Info: Retrieving plugin
Info: Loading facts
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: compile error
/etc/puppet/modules/filebeat/templates/filebeat3.yml.erb:6: syntax error, unexpected ')'
) ).to_s)
 ^
/etc/puppet/modules/filebeat/templates/filebeat3.yml.erb:7: syntax error, unexpected tIDENTIFIER, expecting ')'
_erbout
       ^
Warning: Not using cache on failed catalog
Error: Could not retrieve catalog; skipping run

maybe I'm doing something wrong. Please let me know if this needs any fixing.

**Comment by pcfens** [comment]
What version of the module are you using? Also, are you using the passenger puppetmaster or puppetserver? If you're using passenger, which version of ruby are you using, if you're using puppetserver, which version?

Unfortunately, I'm not able to re-create the issue using puppetserver 1.1.3 or with puppet apply in puppet 3.8.

**Comment by harshal-shah** [comment]
Hi,
I'm using version 0.2.2 of the module.

I'm using puppetserver. Version of puppetserver is

[root@master filebeat]# puppet resource package puppetserver
package { 'puppetserver':
  ensure => '1.1.3-1.el6',
}

The ruby version is :
[root@master filebeat]# gem -v
1.3.7

I really hope I'm not setting any false alarms as I'm still new to the
ecosystem.

Thanks for your reply.

**Comment by pcfens** [comment]
Can you attach /etc/puppet/modules/filebeat/templates/filebeat3.yml.erb from the server? The error message is referencing something that isn't showing up in the latest version.

**Comment by harshal-shah** [comment]
Hi,
Attached.

**Comment by pcfens** [comment]
Hmmm - the file didn't make it through - could you try attaching it via the web interface instead of in an e-mail?

**Comment by adriaaah** [comment]
Hi,

same here with latest version, 0.2.2. I paste my `filebeat3.yml.erb`:

```
<%- @filebeat_config['filebeat']['spool_size'] = Integer(@filebeat_config['filebeat']['spool_size']) -%>
### Filebeat configuration managed by Puppet ###
<%= YAML.dump(@filebeat_config).gsub(/^\s\s|---/,
  '  ' => '',
  '---' => '',
) -%>
```

Hope it helps to debug the issue. BTW, have a nice new year.

PS: I'm running it in Vagrant, so if you prefer I can create a new issue.

**Comment by harshal-shah** [comment]
Hi
I'm using vagrant as well.
and wish  you a very Happy new year.

**Comment by harshal-shah** [comment]
Sorry the files didnt get uploaded from web interface as well :(
Pasting them verbatim here:

-------------filebeat.yml.erb ------------------------
 ### Filebeat configuration managed by Puppet ###
<%= YAML.dump(@filebeat_config).gsub(/---|\s\n/,  "---" => "",  " \n" => "\n") -%>
----------------------- EOF -------------------------------------

--------------------------- filebeat3.yml.erb -----------------------------
<%- @filebeat_config['filebeat']['spool_size'] = Integer(@filebeat_config['filebeat']['spool_size']) -%>
### Filebeat configuration managed by Puppet ###
<%= YAML.dump(@filebeat_config).gsub(/^\s\s|---/,
  '  ' => '',
  '---' => '',
) -%>
-----------------------------------EOF ---------------------------------------------------

---------------------------------- prospector.yml.erb ---------------------------
filebeat:
  prospectors:
    - paths:
      <%- @paths.each do |log_path| -%>
      - <%= log_path %>
      <%- end -%>
      <%- if @encoding -%>
      encoding: <%= @encoding -%>
      <%- end %>
      <%- if @fields.length > 0 -%>
      fields:
        <%- @fields.each do |k, v| -%>
        <%= k %>: <%= v %>
        <%- end -%>
      <%- end -%>
      fields_under_root: <%= @fields_under_root %>
      <%- if @input_type -%>
      input_type: <%= @input_type -%>
      <%- end %>
      <%- if @ignore_older -%>
      ignore_older: <%= @ignore_older -%>
      <%- end %>
      <%- if @log_type -%>
      document_type: <%= @log_type -%>
      <%- end %>
      <%- if @scan_frequency -%>
      scan_frequency: <%= @scan_frequency -%>
      <%- end %>
      <%- if @harvester_buffer_size -%>
      harvester_buffer_size: <%= @harvester_buffer_size -%>
      <%- end %>
      tail_files: <%= @tail_files %>
      force_close_files: <%= @force_close_files %>
      <%- if @backoff -%>
      backoff: <%= @backoff -%>
      <%- end %>
      <%- if @max_backoff -%>
      max_backoff: <%= @max_backoff -%>
      <%- end %>
      <%- if @backoff_factor -%>
      backoff_factor: <%= @backoff_factor -%>
      <%- end %>
      <%- if @partial_line_waiting -%>
      partial_line_waiting: <%= @partial_line_waiting -%>
      <%- end %>

-------------------- EOF ---------------------------------------------------------

**Comment by pcfens** [comment]
It looks like the default ruby version shipped with RHEL 6 didn't get along with the template, and that the latest version of filebeat doesn't get upset with the normal yaml file header (I think it did when I first wrote this, which is all gsub took care of.

Could you all try the fix in the [ruby187 branch](https://github.com/pcfens/puppet-filebeat/tree/ruby187) to see if it works out. If it does I'll test it in my primarily Ubuntu/Debian environment and roll a new release.

**Comment by harshal-shah** [comment]
Hi Phil,
Now the problem is solved :)
Thank you for the fix.

**Comment by pcfens** [comment]
I just released a new version (0.3.0) that should fix this, and simplify things some now that the config file works when YAML has the --- header.

Thanks for submitting this so I could fix it.

**Comment by adriaaah** [comment]
Thank you, also it works for me.
